discovery-provider/src/solana/anchor_program_indexer.py

```diff
@@ -74,23 +80,146 @@ def get_latest_slot(self):
+        if db_models.get("users"):
+            user_ids = list(db_models.get("users", {}).keys())
+            self.msg(f"{user_ids}")
+            session.query(User).filter(
+                User.is_current == True, User.user_id.in_(user_ids)
+            ).update({"is_current": False}, synchronize_session="fetch")
+
+    def extract_ids(self, processed_transactions: List[ParsedTx]):
+        entites: Dict[str, Set[str]] = defaultdict(lambda: set())
+
+        for transaction in processed_transactions:
+            instructions = transaction["tx_metadata"]["instructions"]
+            for instruction in instructions:
+                instruction_name = instruction.get("instruction_name")
+                if instruction_name == "init_admin":
+                    pass
+                elif instruction_name == "update_admin":
+                    pass
+                elif instruction_name == "init_user":
+                    # No action to be taken here
+                    pass
+
+                elif instruction_name == "init_user_sol":
+                    # TODO: parse the tx data for their user id and fetch
+                    # user_id = instruction.get("data").get("id")
+                    pass
+
+                elif instruction_name == "create_user":
+                    # TODO: parse the tx data for their user id and fetch
+                    # user_id = instruction.get("data").get("id")
+                    pass
+
+                elif instruction_name == "update_user":
+                    # TODO: parse the tx data for their user id and fetch
+                    # user_id = instruction.get("data").get("id")
+                    pass
+                elif instruction_name == "update_is_verified":
+                    pass
+                elif instruction_name == "manage_entity":
+                    pass
+                elif instruction_name == "create_content_node":
+                    pass
+                elif instruction_name == "public_create_or_update_content_node":
+                    pass
+                elif instruction_name == "public_delete_content_node":
+                    pass
+                elif instruction_name == "update_user_replica_set":
+                    pass
+                elif instruction_name == "write_entity_social_action":
+                    pass
+                elif instruction_name == "follow_user":
+                    pass
+                elif instruction_name == "init_authority_delegation_status":
+                    pass
+                elif instruction_name == "revoke_authority_delegation":
+                    pass
+                elif instruction_name == "add_user_authority_delegate":
+                    pass
+                elif instruction_name == "remove_user_authority_delegate":
+                    pass
+
+        return entites
+
+    def get_transaction_user_ids(self, processed_transactions: List[Any]) -> Set[str]:
```

Comment:
is the function above supposed to replace this?

Reply:
this will be deleted per offline chat - @jowlee